File: 2/insert_modal.py
import re

# Read the HTML file
with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Read the modal snippet
with open('modal_stats_snippet.html', 'r', encoding='utf-8') as f:
    modal_html = f.read()

# Insert the modal before "<!-- Add Note Modal"
pattern = r'(\s+<!-- Add Note Modal)'
replacement = f'\n\n    {modal_html}\n$1'

new_content = re.sub(pattern, replacement.replace('$1', r'\1'), content, count=1)

# Write back
with open('index.html', 'w', encoding='utf-8') as f:
    f.write(new_content)

print("✅ Modal HTML successfully inserted!")

File: 2/insert_modal.js
const fs = require('fs');

// Read files
const html = fs.readFileSync('index.html', 'utf8');
const modal = fs.readFileSync('modal_stats_snippet.html', 'utf8');

// Insert before "<!-- Add Note Modal"
const marker = '    <!-- Add Note Modal (Redesigned) -->';
const insertion = '\n\n    ' + modal.trim() + '\n\n' + marker;
const newHtml = html.replace(marker, insertion);

// Write back
fs.writeFileSync('index.html', newHtml, 'utf8');

console.log('✅ Modal HTML successfully inserted!');
